ERCS/erc-7550.md

```diff
@@ -0,0 +1,78 @@
+---
+eip: 7550
+title: Transient ERC-20 Approvals
+description: Transient ERC-20 approvals via new contract methods
+author: Alex Kroeger (@akroeger-circle), Joey Santoro (@joeysantoro), Moody Salem (@moodysalem)
+discussions-to: TODO
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-10-26
+requires: 20, 1153
+---
+
+## Abstract
+This standard specifies a single-transaction [ERC-20](./erc-20.md) token approval mechanism as a cheaper and more secure augmentation of standard token approval functionality.
+
+It includes a separate transient (single transaction) mapping of `transientAllowance` which can be set via a call to `transientApprove`.
+
+## Motivation
+
+[ERC-20](./erc-20.md) approvals empower complex smart contract operation of token balances. These are typically set either via [ERC-20](./erc-20.md) `approve` or via [ERC-2612](./erc-2612.md) `permit`.
+
+The existing ERC-20 `allowance` of a token is persistent until revoked or used. This has two drawbacks:
+* gas cost of persistent storage: existing ERC-20 approvals require an `SSTORE` operation, which is relatively expensive, to set an allowance and additionally every time the allowance is modified
+* security: if a bug is exploited in an approved contract that allows an attacker to spend allowed balances in an unxpected way, users' funds can be stolen
+
+[EIP-1153](https://eips.ethereum.org/EIPS/eip-1153) transient storage opcodes enable an alternative approval mechanism which is bound to a single transaction. This simultaneously addresses both issues of existing ERC-20 allowances.
```

Comment:
Consider adding a few more sentences that describes how transient approval helps solve the security issue, and what the transaction flow looks like - to make it more friendly to readers who are new to the concept.